Hunted Mobile/Hunted Mobile/Hunted Mobile/ViewModel/MapViewModel.cs

```diff
@@ -458,24 +461,19 @@ public class MapViewModel : BaseViewModel {
         }
 
         private void ThiefFakePoliceToggle(PlayerEventData data) {
-            Thief updatingPlayer = mapModel.Thiefs.Where(player => player.Id == data.Player.Id).FirstOrDefault();
+            Thief updatingPlayer = mapModel.Thiefs.Where(player => player.Id == data.PlayerBuilder.Id).FirstOrDefault();
             
             if(updatingPlayer == null) {
-                if(data.Player.Id == mapModel.PlayingUser.Id && mapModel.PlayingUser is Thief) {
+                if(data.PlayerBuilder.Id == mapModel.PlayingUser.Id && mapModel.PlayingUser is Thief) {
                     updatingPlayer = mapModel.PlayingUser as Thief;
                 }
             }
 
             if(updatingPlayer != null) {
                 mapModel.Players.Remove(updatingPlayer);
 
-                if(data.Player is FakePolice) {
-                    updatingPlayer = data.Player as FakePolice;
-                }
-                else if(data.Player is Thief) {
-                    updatingPlayer = data.Player as Thief;
-                }
-                else updatingPlayer = new Thief(data.Player, updatingPlayer.CaughtAt);
+                // At this point we know for sure the player is a thief so we can call ToThief
+                updatingPlayer = data.PlayerBuilder.ToThief();
```

Comment:
Is heel het voordeel van een `PlayerBuilder` niet dat de logica voor het teruggeven van een `Police` of `Thief` daarin kan gebeuren? Want wat is anders het voordeel van een `PlayerBuilder` als je erbuiten nog steeds if-statments gaat plaatsen om verschillende uitkomsten eruit te krijgen.

Reply:
Dat is wat er gebeurt in `ToPlayer()`, maar in dit event krijg je een `PlayerBuilder` object dat geen role gegevens heeft (omdat dat niet meegestuurd wordt in het event), maar we weten wel dat dit event nooit om een politieagent gaat. Vandaar dat je er met `ToThief()` alsnog een `Thief` van kan maken zonder dat er een role (`InviteKey`) gegeven is. 
Het had ook opgelost kunnen worden door een `InviteKey` met de thief role toe te voegen en dan `ToPlayer()` te gebruiken. Ik weet niet precies of dat beter was geweest.